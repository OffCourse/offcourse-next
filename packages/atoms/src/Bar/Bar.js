import system from "system-components";
import styled from "styled-components";

const BarWrapper = system(
  {
    display: "flex",
    height: "2.25rem",
    bg: "white",
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    zIndex: 0
  },
  "top",
  "width",
  "bottom",
  "left",
  "right",
  "position",
  "borders",
  "borderColor",
  "space"
);

export default styled(BarWrapper)`
  box-sizing: border-box;
  pointer-events: auto;
  &:hover {
    border-color: ${({ theme }) => theme.colors.primary};
  }
`;
