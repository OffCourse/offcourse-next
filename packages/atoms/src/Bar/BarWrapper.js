import system from "system-components";
import styled from "styled-components";

const BarWrapper = system(
  {
    display: "flex",
    position: "fixed",
    left: 0,
    right: 0,
    borderColor: "grayScale.2",
    bg: "white",
    flex: 1,
    flexDirection: "column"
  },
  "borders",
  "is"
);

export default styled(BarWrapper)`
  box-sizing: border-box;
  pointer-events: auto;
`;
