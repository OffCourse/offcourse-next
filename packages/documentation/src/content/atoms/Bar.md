Bars can be docked at the top...

```react
<div>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Bar position="absolute" docked="top">
    <Heading>Hello World</Heading>
    <Heading>Hello World</Heading>
    <Heading>Hello World</Heading>
  </Bar>
</div>
```

...or the bottom of the window

```react
<div>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Heading>Hello World</Heading>
  <Bar position="absolute" docked="bottom">
    <Heading>Hello World</Heading>
    <Heading>Hello World</Heading>
    <Heading>Hello World</Heading>
  </Bar>
</div>
```

they are fixed by default, but this behaviour can be overridden with the position 'prop'
