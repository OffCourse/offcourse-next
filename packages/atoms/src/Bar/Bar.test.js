import '../setupTests';

import React from 'react';
import ReactDOM from 'react-dom';
import { create as render } from 'react-test-renderer'
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Bar from './BarWrapper';

describe('Bar', () => {
  const props = {
    docked: "bottom",
    position: "static",
  };

  it ('passes props', () => {
    const tree = shallow(<Bar {...props} />);

    expect(tree.props()).toMatchObject(props);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Bar {...props} />);

    expect(toJson(tree)).toMatchSnapshot();
  });
});
