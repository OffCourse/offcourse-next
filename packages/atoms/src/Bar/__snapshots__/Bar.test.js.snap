// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: static;
  left: 0px;
  right: 0px;
  background-color: white;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-right: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2rem;
  z-index: 2;
  box-sizing: border-box;
}

<div
  alignItems="center"
  bg="white"
  blacklist={
    Array [
      "display",
      "justifyContent",
      "justify",
      "position",
      "left",
      "right",
      "color",
      "bg",
      "flexDirection",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "alignItems",
      "align",
      "height",
      "zIndex",
      "top",
      "bottom",
      "css",
    ]
  }
  className="c0"
  display="flex"
  docked="bottom"
  flexDirection="row"
  height="2rem"
  justifyContent="space-between"
  left={0}
  position="static"
  pr={4}
  right={0}
  zIndex={2}
/>
`;
